### PR TITLE
Compiler CMake support for POWER platforms

### PR DESCRIPTION
**Pull request by mgaudet**
As well as miscellaneous refactoring derived from said work.

### CODE REVIEW

**Comment by charliegracie** [comment]
@dnakamura and @rwy0717 ?

File: cmake/compiler_support.cmake [hunk 1]
```diff
@@ -72,19 +71,65 @@ macro(tr_detect_system_information)
...
+		
+	elseif(OMR_HOST_OS MATCHES "aix")
+		list(APPEND TR_COMPILE_DEFINITIONS
+			${upcase_os}
+			SUPPORTS_THREAD_LOCAL
+			_XOPEN_SOURCE_EXTENDED=1 
+    			_ALL_SOURCE
+		)
 	else()
 		jit_not_ready()
```
Comment (dnakamura) [inline cmake/compiler_support.cmake]: maybe an `omr_assert(FALSE, .... )` would be more clear here?

**Comment by dnakamura** [comment]
LGTM. Had small note re: using `omr_assert` but I'm also ok merging in as is

**Comment by mgaudet** [comment]
`OMR_COMPILER_COMPILE_OPTIONS`? Screaming awkward. Historically the compiler technology has been commonly referred to by the name Testarossa, and I don't feel a particular need to wander from that particular heritage.  

The isolation here shouldn't be permanent though: Longer term, it ought to be better integrated into the rest of the compiler technology.

**Comment by mgaudet** [comment]
A lot of the indenting issues pre-date this change. I will  apply a whitespace correction change separately.

**Comment by mgaudet** [comment]
Addressed Robert's review comments in the above trio of commits.

**Comment by rwy0717** [comment]
LGTM

**Pull request merged**